src/components/PupilComponents/UnitsSection/UnitsSection.tsx

```diff
@@ -0,0 +1,113 @@
+import {
```

Comment:
76% coverage on this file at the moment. It would be better to separate it out and have separate tests for this, but think it's better to merge first. I've not really removed tests in this refactor so I think it's okay.